src/System.Runtime/tests/System/TypeTests.netcoreapp.cs

```diff
@@ -36,5 +36,42 @@ public void IsSZArray(Type type, bool expected)
         {
             Assert.Equal(expected, type.IsSZArray);
         }
+
+        private static IEnumerable<object[]> VariableBoundArrayOrNotTypes()
+        {
+            // Not arrays
+            yield return new object[] { typeof(void), false };
+            yield return new object[] { typeof(int), false };
+            yield return new object[] { typeof(int[]).MakeByRefType(), false };
+            yield return new object[] { typeof(TypeTests), false };
+            yield return new object[] { Type.GetType("System.Int32"), false };
+            yield return new object[] { typeof(Outside<int>.Inside<string>), false };
+
+            // Arrays, but SZ arrays
+            yield return new object[] { Type.GetType("System.Int32[]"), false };
+            yield return new object[] { typeof(int[]), false };
```

Comment:
We might as make sure all the type flavors are represented here.

In particular, we could add:

 - Pointer types (typeof(int*))

 - Generic type variables

    typeof(GenericClass<>).GetTypeInfo().GenericTypeParameters[0]

 - Generic type variables for methods
 
    MethodInfo m = ...some generic method..
    m.GetGenericArguments()[0]

 - The "weird" types created from CLSID's:

   GUID clsid = default(GUID);
   Type.GetTypeFromCLSID(clsid);

  - Internal framework types blocked from Reflection on .NETNative

     new object().GetType().GetType()     (Yes, it looks weird - don't worry about it :-)